Cubical/Data/FinWeak/Base.agda

```diff
@@ -0,0 +1,57 @@
+{-# OPTIONS --safe #-}
+module Cubical.Data.FinWeak.Base where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Function
+
+open import Cubical.Data.Bool
+open import Cubical.Data.Nat using (ℕ; zero; suc)
+
+open import Cubical.Relation.Nullary
+
+private variable
+  n : ℕ
+
+data FinPure : ℕ → Type where
+  zero : FinPure 1
+  suc  : FinPure n → FinPure (suc n)
+
```

Comment:
The definition of ```FinPure``` looks almost exactly like ```Fin``` in ```Cubical.Data.FinData.Base```:
```agda
data Fin : ℕ → Type₀ where
  zero : {n : ℕ} → Fin (suc n)
  suc  : {n : ℕ} (i : Fin n) → Fin (suc n)
```
A comment describing the crucial difference would help a reader here. It would also be good to have some comment on use cases or the meaning of ```FinPure``` and the hint that it turns out to be a proposition.